client/disputes/evidence/index.js

```diff
@@ -376,19 +376,25 @@ export default ( { query } ) => {
 			'There are unsaved changes on this page. Are you sure you want to leave and discard the unsaved changes?',
 			'woocommerce-payments'
 		);
-	}, [ pristine ] );
+	} );
+
+	useEffect( confirmationNavigationCallback, [
+		pristine,
+		confirmationNavigationCallback,
+	] );
 
-	const fetchDispute = async () => {
```

Comment:
Moved inside the `useEffect` hook - otherwise it would have needed to be declared as dependency and stabilized with `useCallback`